ImagePicker/ImagePickerController.swift

```diff
@@ -1,154 +1,115 @@
-//
-//  ImagePickerController.swift
-//  Image Picker
-//
-//  Created by Peter Stajger on 04/09/2017.
-//  Copyright © 2017 Inloop. All rights reserved.
-//
+// Copyright © 2018 INLOOPX. All rights reserved.
 
 import Foundation
-import UIKit
 import Photos
 
-open class ImagePickerController : UIViewController {
-    deinit {
-        PHPhotoLibrary.shared().unregisterChangeObserver(self)
-        captureSession?.suspend()
-        log("deinit: \(String(describing: self))")
-    }
-    
-    // MARK: - Public API
-
-    ///
+open class ImagePickerController: UIViewController {
     /// Use this object to configure layout of action, camera and asset items.
-    ///
     public var layoutConfiguration = LayoutConfiguration.default
     
-    ///
     /// Use this to register a cell classes or nibs for each item types
-    ///
     public lazy var cellRegistrator = CellRegistrator()
     
-    ///
     /// Use these settings to configure how the capturing should behave
-    ///
     public var captureSettings = CaptureSettings.default
     
-    ///
     /// Get informed about user interaction and changes
-    ///
     public weak var delegate: ImagePickerControllerDelegate?
     
-    ///
     /// Provide additional data when requested by Image Picker
-    ///
     public weak var dataSource: ImagePickerControllerDataSource?
     
-    ///
+    /// Access all currently selected images
+    public var selectedAssets: [PHAsset] {
+        let selectedIndexPaths = collectionView.indexPathsForSelectedItems ?? []
+        let selectedAssets = selectedIndexPaths.compactMap { indexPath in
+            return asset(at: indexPath.row)
+        }
+        return selectedAssets
+    }
+    
+    /// Fetch result of assets that will be used for picking.
+    /// If you leave this nil or return nil from the block, assets from recently
+    /// added smart album will be used.
+    public var assetsFetchResultBlock: (() -> PHFetchResult<PHAsset>?)?
+
+    /// A collection view that is used for displaying content.
+    public var collectionView: UICollectionView! {
+        return imagePickerView.collectionView
+    }
+    
+    private var collectionViewCoordinator: CollectionViewUpdatesCoordinator!
+    private var imagePickerView: ImagePickerView! {
```

Comment:
not a good idea to use force unwrapping even it is in the framework itself.

Reply:
Yes, but i dont see a way how to recover from this, its better to crash at this line than have many checks when variable is used.